### PR TITLE
Training page to display training details

### PR DESCRIPTION
**Pull request by christina-mcmullan**
Added Training route to apiService to gain access to Training page.
Added TrainingDetails.html to display training information on the webpage.
Added trainingData.js  to get training data according to BandTier

### CODE REVIEW

File: application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java [hunk 1]
```diff
@@ -107,6 +107,83 @@ public static boolean getConnection(){
...
+                if(bandName == "Foundational– Associate" || bandName.equals("foundational– associate")) {
+                    BandID = 10;
+                }
+                if(bandName == "Developing–Senior Associate" || bandName.equals("developing–senior associate")) {
+                    BandID = 11;
+                }
+                if(bandName == "Intermediate–Consultant" || bandName.equals("intermediate–consultant")) {
+                    BandID = 12;
+                }
+                if(bandName == "Beginner-Apprentice" || bandName.equals("beginner-apprentice")) {
```
Comment (John-Neill-Kainos) [inline application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java:136]: ensure that there is no spaces, replace space with "-" e.g developing senior associate -> developing-senior-associate

File: application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java [hunk 2]
```diff
@@ -107,6 +107,83 @@ public static boolean getConnection(){
...
+
+
+
+
+
+
+
+
+
+
```
Comment (John-Neill-Kainos) [inline application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java:172]: remove all the white space

**Review by liammc98 — COMMENTED** [review]
It would be best if you change the subsequent if() statements to else if() for your API call, means we're not checking things unnecessarily 

Should also remove the additional whitespace below the API method call

File: application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java [hunk 3]
```diff
@@ -107,6 +107,83 @@ public static boolean getConnection(){
...
+                queryInformation.put("Error - Invalid path entered");
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+
+        return ResponseEntity.ok(queryInformation.toString());
+    }
+
+
```
Comment (liammc98) [inline application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java:163]: Whitespace is here, best to remove it

File: Views/trainingData.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const fetch = require("node-fetch");
+
+exports.getTrainingData = async (bandName) => {
+    const apiRoute = 'http://localhost:8080/roles/' + bandName + '/training';
+    const response = await fetch(apiRoute, {method: 'GET', headers:{} })
+    const data = await response.json();
+    //console.log(data)
```
Comment (jkelso662) [inline Views/trainingData.js:7]: Remove commented-out code. Thanks

File: application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java [hunk 5]
```diff
@@ -108,6 +108,60 @@ public static boolean getConnection(){
...
+                }
+                else if(bandName.equals("developing-senior-associate")) {
+                    BandID = 11;
+                }
+                else if(bandName.equals("intermediate-consultant")) {
+                    BandID = 12;
+                }
+                else if(bandName.equals("beginner-apprentice"))  {
+                    BandID = 13;
+                }
```
Comment (jkelso662) [inline application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java:138]: I think a switch statement would make more sense here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

File: application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java [hunk 6]
```diff
@@ -108,6 +108,60 @@ public static boolean getConnection(){
...
+        ResultSet preJsonResultSet = null;
+        int BandID = 0;
+
+        if(conn == null){
+            queryInformation.put("Error no connection created");
+            return ResponseEntity.ok(queryInformation.toString());
+        }
+
+        try {
+            if (bandName != null && !bandName.isEmpty()){
```
Comment (jkelso662) [inline application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java:123]: For future reference, `if(variable){...}` is equivalent to `if(variable != undefined)` and  `if(variable != null){...}`

**Review by jkelso662 — COMMENTED** [review]
Also some more white space issues again would like to see cleaned up. Also I noticed there is a `.DS_Store` file in the views folder would need removed and ignored. Happy to approve when the majority of these other things are sorted. Good work!

File: routes.js [hunk 7]
```diff
@@ -116,4 +114,20 @@ router.get('/capability/:capabilityName', async function (req, res) {
         console.log(ex);
     }
 });
+
+
+router.get('/roles/:bandName/training', async function (req, res) {
+
+
+ try{
```
Comment (jkelso662) [inline routes.js:122]: Lots of blank space, and also incorrect indentation.

File: application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java [hunk 8]
```diff
@@ -382,6 +382,48 @@ public static JSONArray getJobFamilyUsingCapabilityID(String capabilityID){
...
+        int BandID = 0;
+
+        if(conn == null){
+            queryInformation.put("Error no connection created");
+            return ResponseEntity.ok(queryInformation.toString());
+        }
+
+        try {
+            if (bandName != null){
+                if(bandName.equals("basic-trainee")){ BandID= 9; }
```
Comment (jkelso662) [inline application/apiService/src/main/java/com/kainos/ea/apiService/controllers/apiService.java:398]: https://github.com/liammc98/KainosJobsPortal/pull/17#discussion_r672286847

**Review by liammc98 — APPROVED** [review]

**Review by John-Neill-Kainos — APPROVED** [review]

**Pull request merged**